server/src/main/java/org/opensearch/cluster/routing/allocation/decider/ShardsLimitAllocationDecider.java

```diff
@@ -125,6 +126,15 @@ private Decision doDecide(
         RoutingAllocation allocation,
         BiPredicate<Integer, Integer> decider
     ) {
+        /*
+         Remote shards were introduced as a part of searchable snapshots. The following block
+         enables allocation overrides for remote shards since they do not use the local storage as
+         the primary source of data storage.
+         */
+        if (RoutingPool.REMOTE_CAPABLE.equals(RoutingPool.getNodePool(node))
```

Comment:
same for this one, same logic in DiskThresholdDecider, it would be better if we can reuse same logic maybe if as simple as a static method in some util class